pybind11_stubgen/parser/mixins/fix.py

```diff
@@ -43,22 +43,35 @@ class RemoveSelfAnnotation(IParser):
     def handle_method(self, path: QualifiedName, method: Any) -> list[Method]:
         methods = super().handle_method(path, method)
         for method in methods:
-            self._remove_self_arg_annotation(method.function)
+            self._remove_self_arg_annotation(path, method.function)
         return methods
 
     def handle_property(self, path: QualifiedName, prop: Any) -> Property | None:
         prop = super().handle_property(path, prop)
         if prop is not None:
             if prop.getter is not None:
-                self._remove_self_arg_annotation(prop.getter)
+                self._remove_self_arg_annotation(path, prop.getter)
             if prop.setter is not None:
-                self._remove_self_arg_annotation(prop.setter)
+                self._remove_self_arg_annotation(path, prop.setter)
 
         return prop
 
-    def _remove_self_arg_annotation(self, func: Function):
-        if len(func.args) > 0 and func.args[0].name == "self":
-            func.args[0].annotation = None
+    def _remove_self_arg_annotation(self, path: QualifiedName, func: Function) -> None:
+        if len(func.args) == 0:
+            return
+        path = QualifiedName(path[:-1])  # remove method name from path
+        first_arg = func.args[0]
+        if (
+            first_arg.name == "self"
+            and isinstance(first_arg.annotation, ResolvedType)
+            and (
+                first_arg.annotation.name == QualifiedName.from_str("Any")
+                or first_arg.annotation.name == QualifiedName.from_str("typing.Any")
+                or first_arg.annotation.name == path
+                or first_arg.annotation.name == path[-len(first_arg.annotation.name) :]
```

Comment:
let's change it to smth like `first_arg.annotation.name in {any_t_name, full_any_t_name, self_class_name, full_self_class_name}`

Let's also check for empty parameters of `ResolvedType`

Reply:
Would you be interested in turning all of the `QualifiedName.from_str(<static string>)` into constants at the top of this file?

Reply:
Is it preferable to do `ResolvedType(name=TYPING_ANY_T_NAME)` than `self.parse_annotation_str("Any")`? I think the obvious benefit is reusing the constant. What do you think?

Reply:
Let's make them class private constants like in other "fix" classes.